case-lib/lib.sh

```diff
@@ -752,17 +752,84 @@ func_lib_check_pa()
 # However, 1. arrays would complicate the user interface 2. ALSA does not
 # seem to need arguments with whitespace or globbing characters.
 
+# SOF_ALSA_TOOL:
+# This option is used for selecting tool for testing,
+# So far, supported tools are 'alsa' and 'tinyalsa'
+# To select appropriate tool, set SOF_ALSA_TOOL to one of above
+# before using 'aplay_opts' or 'arecord_opts' function.
+# (e.g., SOF_ALSA_TOOL=alsa)
+
+
+# Function to extract the card number and device number from $dev option (e.g., hw:0,10)
+parse_audio_device() {
+    # Extract the card number (e.g., "0" from hw:0,10)
+    card_nr=$(printf '%s' "$dev" | cut -d ':' -f2 | cut -d ',' -f1)
+
+    # Extract the device number (e.g., "10" from hw:0,10)
+    dev_nr=$(printf '%s' "$dev" | cut -d ',' -f2)
+}
+
+# Function to extract the numeric format value from the $fmt_elem option
+extract_format_number() {
+    # (e.g., extracting '16' from 'S16_LE')
+    # shellcheck disable=SC2154
+    format=$(printf '%s' "$fmt_elem" | grep '[0-9]\+' -o)
+}
+
+# Initialize the parameters using for audio testing.
+initialize_audio_params()
+{
+   # shellcheck disable=SC2034
+   channel=$(func_pipeline_parse_value "$idx" channel)
+   # shellcheck disable=SC2034
+   rate=$(func_pipeline_parse_value "$idx" rate)
+   # shellcheck disable=SC2034
+   fmts=$(func_pipeline_parse_value "$idx" fmt)
+   # shellcheck disable=SC2034
+   dev=$(func_pipeline_parse_value "$idx" dev)
+   # shellcheck disable=SC2034
+   pcm=$(func_pipeline_parse_value "$idx" pcm)
+   # shellcheck disable=SC2034
+   type=$(func_pipeline_parse_value "$idx" type)
+   # shellcheck disable=SC2034
+   snd=$(func_pipeline_parse_value "$idx" snd)
+
+   : "${SOF_ALSA_TOOL:="alsa"}"
+   if [[ "$SOF_ALSA_TOOL" = "tinyalsa" ]]; then
+      parse_audio_device
+   fi
+}
+
 aplay_opts()
 {
-    dlogc "aplay $SOF_ALSA_OPTS $SOF_APLAY_OPTS $*"
-    # shellcheck disable=SC2086
-    aplay $SOF_ALSA_OPTS $SOF_APLAY_OPTS "$@"
+    if [[ "$SOF_ALSA_TOOL" = "tinyalsa" ]]; then
+        dlogc "tinyplay $*"
+	# shellcheck disable=SC2154
```

Comment:
I don't see this warning. Did you use the `-x` option? `shellcheck -x ...`

Reply:
I think I know what's happening! Cause it just happened to me in a totally unrelated context. You must run `shellcheck -x case-lib/lib.sh` IN THE TOP-LEVEL sof-test directory! That will hopefully silence a few warnings.